payment-guarantee/src/contract.rs

```diff
@@ -0,0 +1,698 @@
+use std::borrow::BorrowMut;
+
+use cosmwasm_std::{attr, Binary, BlockInfo, Coin, coin, Deps, DepsMut, Empty, Env, from_binary, MessageInfo, Order, Pair, Response, StdError, StdResult, to_binary};
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cw721::{Cw721QueryMsg, Cw721ReceiveMsg, Expiration, NftInfoResponse, NumTokensResponse, TokensResponse};
+use cw721_base::ContractError;
+use cw721_base::msg::{InstantiateMsg, MinterResponse};
+
+use crate::msg::{ExecuteMsg, MintMsg, QueryMsg};
+use crate::query::{NftInfoWithPriceResponse, TokenPriceResponse};
+use crate::receiver::NftInfoWithPriceMsg;
+use crate::state::TOKEN_PRICE;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
```

Comment:
Most use the functionality of `cw721_base`.
Implement only for additional needs.

Reply:
I don't understand what you mean. 
Do you mean that `The 'instantiate()' function is mostly used by the 'cw721_base'`?

Reply:
Ah. I understood that this `instantiate()` function just calls the `cw721_base::contract::instantiate()` function as it is.

Reply:
Yes it is.